tests/raster/test_geotiff.py

```diff
@@ -0,0 +1,307 @@
+
+    def test_read_write_specific_band(self):
+        """
+        Test a write and read of a specific mosaic.
+        """
+        data = np.ones((100, 100), dtype=np.float32)
+
+        with GeoTiffFile(self.filepath, mode='w', n_bands=10) as src:
+            src.write({5: data})
+            src.write({10: data})
+
+        with GeoTiffFile(self.filepath) as src:
+            ds = src.read(bands=5)
+            np.testing.assert_array_equal(ds[5], data)
+            ds = src.read(bands=10)
+            np.testing.assert_array_equal(ds[10], data)
+
+    def test_metadata(self):
+        """
+        Test meta mosaic tags.
+        """
+        data = np.ones((1, 100, 100), dtype=np.float32)
+
+        metadata = {'attr1': '123', 'attr2': 'test'}
+
+        with GeoTiffFile(self.filepath, mode='w', metadata=metadata) as src:
+            src.write(data)
+
+        with GeoTiffFile(self.filepath) as src:
+            ds_md = src.metadata
+
+        self.assertEqual(ds_md, metadata)
+
+    def test_geotransform(self):
+        """
+        Test geotransform and spatialref keywords.
+
+        Notes
+        -----
+        Spatial reference cannot be compared due to different versions of WKT1 strings.
+
+        """
+        data = np.ones((1, 100, 100), dtype=np.float32)
+
+        geotrans_ref = (3000000.0, 500.0, 0.0, 1800000.0, 0.0, -500.0)
+        sref = ('PROJCS["Azimuthal_Equidistant",GEOGCS["WGS 84",'
+                     'DATUM["WGS_1984",SPHEROID["WGS 84",6378137,'
+                     '298.257223563,AUTHORITY["EPSG","7030"]],'
+                     'AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0],'
+                     'UNIT["degree",0.0174532925199433],'
+                     'AUTHORITY["EPSG","4326"]],'
+                     'PROJECTION["Azimuthal_Equidistant"],'
+                     'PARAMETER["latitude_of_center",53],'
+                     'PARAMETER["longitude_of_center",24],'
+                     'PARAMETER["false_easting",5837287.81977],'
+                     'PARAMETER["false_northing",2121415.69617],'
+                     'UNIT["metre",1,AUTHORITY["EPSG","9001"]]]')
+
+        with GeoTiffFile(self.filepath, mode='w', geotrans=geotrans_ref, sref_wkt=sref) as src:
+            src.write(data)
+
+        with GeoTiffFile(self.filepath) as src:
+            geotrans_val = src.geotrans
+
+        self.assertEqual(geotrans_val, geotrans_ref)
+
+    def tearDown(self):
+        """
+        Remove test file.
+        """
+        os.remove(self.filepath)
+
+
+class GeoTiffDataTest(unittest.TestCase):
+    """
+    Testing a GeoTIFF image stack.
+    """
+
+    def setUp(self):
+        """
+        Set up dummy mosaic set.
+        """
+        self.path = mkdtemp()
+        if not os.path.isdir(self.path):
+            os.makedirs(self.path)
+
+    def test_write_read_image_stack(self):
+        """
+        Test writing and reading an image stack.
+        """
```

Comment:
doc strings should be used for outward facing public APIs, test functions should have descriptive names. In this case the doc string is superfluous as the test name already tells the whole story.

Also see this short article on necessary comments in general: http://blog.cleancoder.com/uncle-bob/2017/02/23/NecessaryComments.html

Reply:
Done.